mev-build-rs/src/payload/builder.rs

```diff
@@ -95,10 +96,26 @@ fn append_payment<Client: StateProviderFactory>(
 
     let signer_account = db.load_cache_account(signer.address())?;
     let nonce = signer_account.account_info().map(|info| info.nonce).unwrap_or_default();
+
+    let proposer_fee_recipient_account = db.load_cache_account(config.proposer_fee_recipient)?;
+    let is_empty_code_hash =
+        proposer_fee_recipient_account.account_info().unwrap_or_default().is_empty_code_hash();
+
+    // If the proposer (fee recipient) has associated code, we hardcode 250_000 for gas payment.
+    // If it is a regular EOA, we send the base gas 21_000.
```

Comment:
and to reduce future churn, we can be more abstract in the comment, something like:

```suggestion
    // Use a fixed gas limit for the payment transaction reflecting the recipient's status as smart contract or EOA.
```